src/globus_sdk/services/search/data.py

```diff
@@ -221,6 +224,49 @@ def add_sort(
         return self
 
 
+class SearchQueryV1(utils.PayloadWrapper):
+    """
+    A specialized dict which has helpers for creating and modifying a Search
+    Query document. Replaces the usage of ``SearchQuery``.
+
+    :param q: The query string. Required unless filters are used.
+    :param limit: A limit on the number of results returned in a single page
+    :param offset: An offset into the set of all results for the query
+    :param advanced: Whether to enable (``True``) or not to enable (``False``) advanced
+        parsing of query strings. The default of ``False`` is robust and guarantees that
+        the query will not error with "bad query string" errors
+    :param additional_fields: additional data to include in the query document
+    """
+
+    def __init__(
+        self,
+        q: str | None = None,
+        *,
+        limit: int | None = None,
+        offset: int | None = None,
+        advanced: bool | None = None,
```

Comment:
I believe that all of these should have defaults of `MISSING`, which will allow us to set them in the body without `if`-checks on their values as well.

We recently discussed this over on the Compute-related PR, so I think the explanation I wrote there is really solid and explains the `None` vs `MISSING` divide:
https://github.com/globus/globus-sdk-python/pull/1078#discussion_r1794192893

But we can discuss here if there's anything unclear in that.

Reply:
This was marked resolved but I still see `... | None = None` in the diff, so I'm unresolving.